src/registries/npm-registry.js

```diff
@@ -23,6 +23,12 @@ const DEFAULT_REGISTRY = 'https://registry.npmjs.org/';
 const REGEX_REGISTRY_PREFIX = /^https?:/;
 const REGEX_REGISTRY_SUFFIX = /registry\/?$/;
 
+// All scoped package names are of the format `@scope%2fpkg` from the use of NpmRegistry.escapeName
+// `(?:^|\/)` Match either the start of the string or a `/` but don't capture
+// `[^\/?]*?` Match any character that is not '/' or '?' and capture, up until the first occurance of:
+// `%2f` Match '%2f' the escaped forward slash and don't capture
+const SCOPED_PKG_REGEXP = /(?:^|\/)(@[^\/?]*?)(?=%2f)/;
```

Comment:
I wasn't sure if a eg `@scope%2ffoo%2fbar` should have scope `@scope` or `@scope%2ffoo` so your knowledge is appreciated there.